lib/pbench/server/api/resources/query_apis/datasets_publish.py

```diff
@@ -179,59 +178,80 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
+                        "shard": "0",
+                        "index": "index1"
                     }
-                },
+                }
+            },
+
+        Return a summary of success and failure operations across the set of
+        indices, for example:
+
+            {
+                "test.v6.run-data.2021-07": { "success": 1, "failure": 0 },
+                "test.v6.run-toc.2021-07": { "success": 500, "failure": 4 }
+            }
         """
+        report = {}
         error_count = 0
-        if es_json["errors"]:
-            for i in es_json["items"]:
-                u = i["update"]
-                if u["status"] != HTTPStatus.OK:
-                    e = u["error"]
-                    self.logger.warning(
-                        "{} ({}: {}) for id {} in index {}",
-                        u["status"],
-                        e["type"],
-                        e["reason"],
-                        u["_id"],
-                        u["_index"],
-                    )
-                    error_count += 1
+        for i in es_json["items"]:
+            u = i["update"]
+            index = u["_index"]
+            if index not in report:
+                report[index] = {"success": 0, "failure": 0}
+            if u["status"] == HTTPStatus.OK:
+                report[index]["success"] += 1
+            else:
+                report[index]["failure"] += 1
+                e = u["error"]
+                self.logger.warning(
+                    "{} ({}: {}) for id {} in index {}",
+                    u["status"],
+                    e["type"],
+                    e["reason"],
+                    u["_id"],
+                    u["_index"],
+                )
+                error_count += 1
+
+        dataset: Dataset = context["dataset"]
+        self.logger.info(
+            "Update access for dataset {}: {} indices, {} successful document updates and {} failures",
+            dataset,
+            len(report),
+            len(es_json["items"]) - error_count,
+            error_count,
+        )
+
         if error_count > 0:
-            abort(
+            raise PostprocessError(
                 HTTPStatus.INTERNAL_SERVER_ERROR,
-                message=(
+                (
                     f"{error_count:d} of {context['document_count']:d} "
                     "Elasticsearch document UPDATE operations failed"
                 ),
             )
         else:
             # Only on total success we update the Dataset's registered access
             # column; a "partial success" will remain in the previous state.
-            # TODO: Is it worth dealing with a third "INCOMPLETE_PUBLISH" state
-            # to record this? (Or a separate error state column that could be
-            # tracked and reported?)
-            dataset: Dataset = context["dataset"]
             dataset.access = context["access"]
             dataset.update()
 
-        return "", HTTPStatus.OK
+        return report
```

Comment:
OK, I'd meant to go a bit further here and work out a way to return the `report` document on failure. I'm not really sure how that concept fits in with the `abort` protocol, and I forgot to investigate and experiment before pushing a commit. I may work on that more tomorrow, since the `report` I'm returning now is just a count of documents with 0 errors. Interesting, but not all I'd been going for...